lib/providers/env_var_provider.ex

```diff
@@ -209,16 +226,12 @@ defmodule EnvVar.Provider do
     |> Enum.map(&convert(&1, type))
   end
 
-  defp set_default(value, default, _env_var_name) when is_nil(value) do
+  defp set_default(value, default) when is_nil(value) or value == "" do
```

Comment:
Elixir can tell the difference between an env var that is set, but the value is `""` and when it's not set at all (value of `nil`). We should NOT combine those behaviors.

e.g. 

```
$ asdf="" iex
...
iex(1)> System.get_env "asdf"
""
```

Reply:
Looks like we may previously have treated them the same way, but we shouldn't.

Reply:
> Elixir can tell the difference between an env var that is set, but the value is "" and when it's not set at all (value of nil)

Oh I know, this change was just a simpler refactor of the previous behaviour. I agree we should treat them differently. Just to confirm, you want an empty string to be an empty string and not to fall back to the default value, correct?

Reply:
Yeah, that seems like the right behavior.